packages/function-tree/src/index.test.js

```diff
@@ -0,0 +1,37 @@
+/* eslint-env mocha */
+import FunctionTreeExecution from './'
+
+describe('FunctionTreeExecution', () => {
+  describe('execution.abort()', () => {
+    it('should stop if aborted from sync action', (done) => {
+      const runTree = FunctionTreeExecution([])
+      runTree.on('abort', () => done())
+
+      runTree([
+        ({execution}) => {
+          return execution.abort()
+        },
+        ({execution}) => {
+          done(new Error('Should not continue execution.'))
+        }
+      ])
+    })
+    it('should stop if aborted from async action', (done) => {
+      const runTree = FunctionTreeExecution([])
+      runTree.on('abort', () => done())
+
+      runTree([
+        ({execution}) => {
+          return new Promise((resolve) => {
+            setTimeout(() => {
```

Comment:
Actually does not need timeout, promises are async in nature, so you can just return:

```js
  return Promise.resolve(execution.abort())
```